ui-events/src/pointer/buttons.rs

```diff
@@ -100,32 +134,59 @@ impl PointerButtons {
 
 impl core::fmt::Debug for PointerButtons {
     fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
-        let mut tuple = f.debug_tuple("PointerButtons");
-        if self.contains(PointerButton::Primary) {
-            tuple.field(&"Primary");
-        }
-        if self.contains(PointerButton::Secondary) {
-            tuple.field(&"Secondary");
-        }
-        if self.contains(PointerButton::Auxiliary) {
-            tuple.field(&"Auxiliary");
-        }
-        if self.contains(PointerButton::X1) {
-            tuple.field(&"X1");
-        }
-        if self.contains(PointerButton::X2) {
-            tuple.field(&"X2");
-        }
-        if self.contains(PointerButton::PenEraser) {
-            tuple.field(&"PenEraser");
+        if self.is_empty() {
+            return f.write_str("PointerButtons(None)");
         }
-        if self.contains(PointerButton::Other) {
-            tuple.field(&"Other");
+
+        f.write_str("PointerButtons(")?;
+
+        let mut first = true;
+        for &button in &[
+            PointerButton::Primary,
+            PointerButton::Secondary,
+            PointerButton::Auxiliary,
+            PointerButton::X1,
+            PointerButton::X2,
+            PointerButton::PenEraser,
+            PointerButton::B7,
+            PointerButton::B8,
+            PointerButton::B9,
+            PointerButton::B10,
+            PointerButton::B11,
+            PointerButton::B12,
+            PointerButton::B13,
+            PointerButton::B14,
+            PointerButton::B15,
+            PointerButton::B16,
+            PointerButton::B17,
+            PointerButton::B18,
+            PointerButton::B19,
+            PointerButton::B20,
+            PointerButton::B21,
+            PointerButton::B22,
+            PointerButton::B23,
+            PointerButton::B24,
+            PointerButton::B25,
+            PointerButton::B26,
+            PointerButton::B27,
+            PointerButton::B28,
+            PointerButton::B29,
+            PointerButton::B30,
+            PointerButton::B31,
+            PointerButton::B32,
+        ] {
+            if self.contains(button) {
+                if !first {
+                    f.write_str(" | ")?;
```

Comment:
I *think* that if the alternate mode is on, this should be `\n    | `. However, it's sufficiently unlikely that there would be enough buttons enabled that that would make a difference

Reply:
Is that how it's meant to work even when they could be nested?

Reply:
Yes - the parent DebugStruct intercepts any newlines and replaces them with `\n    `. It's a neat trick, which luckily we don't have to bother with reproducing.

Reply:
Alternatively, the alternate formatting could be just the number. But idk, I don't think it's worth caring about too much

Reply:
I've added some alternate debug formatting stuff, it's not perfect but it's better than nothing.